tools/adventure-pack/goodies/typescript/Iterator.prototype.toArray/index.test.ts

```diff
@@ -0,0 +1,24 @@
+import "../Iterator.prototype.toArray";
+import { describe, expect, it } from "@jest/globals";
+
+describe("Iterator.prototype.toArray", () => {
+  it("can convert an iterator to an array", () => {
+    expect(
+      [1, 1, "a", 2, "b", "b", 3, "c"][Symbol.iterator]().toArray(),
+    ).toEqual([1, 1, "a", 2, "b", "b", 3, "c"]);
+  });
+
+  it("can return an empty array for an empty iterator", () => {
+    expect([].values().toArray()).toEqual([]);
+  });
+
+  it("can chain with other iterator methods", () => {
+    const iterator = [1, 2, 3, 4, 5].values();
+    expect(
+      iterator
+        .toArray()
+        .map((x) => x ** 2)
+        .filter((x) => x % 2 === 0),
```

Comment:
I like this test, but after the change `toArray`, these are no longer iterator methods, they are array methods. Is there another way to do this test?

Reply:
Ack!